tests/Unit/NumericalAlgorithms/DiscontinuousGalerkin/Test_MetricIdentityJacobian.cpp

```diff
@@ -0,0 +1,160 @@
+auto make_map() noexcept;
+
+template <>
+auto make_map<1>() noexcept {
+  // Works when identity...
+  return domain::make_coordinate_map<Frame::Logical, Frame::Inertial>(
+      Affine{-1.0, 1.0, -0.3, 0.7});
+}
+
+template <>
+auto make_map<2>() noexcept {
+  return domain::make_coordinate_map<Frame::Logical, Frame::Inertial>(
+      Affine2D{{-1.0, 1.0, -1.0, -0.99}, {-1.0, 1.0, -1.0, -0.99}},
+      domain::CoordinateMaps::Wedge2D{1.0, 2.0, 0.0, 1.0, {}, false});
+}
+
+template <>
+auto make_map<3>() noexcept {
+  return domain::make_coordinate_map<Frame::Logical, Frame::Inertial>(
+      Affine3D{{-1.0, 1.0, -1.0, -0.99},
+               {-1.0, 1.0, -1.0, -0.99},
+               {-1.0, 1.0, -1.0, 1.0}},
+      domain::CoordinateMaps::ProductOf2Maps<domain::CoordinateMaps::Wedge2D,
+                                             Affine>{
+          {1.0, 2.0, 0.0, 1.0, {}, false}, {0.0, 1.0, 0.0, 1.0}});
+}
+
+
+template <size_t Dim>
+void test(const Mesh<Dim>& mesh) {
+  CAPTURE(Dim);
+  CAPTURE(mesh);
+  std::uniform_real_distribution<double> dist(-1.0, 2.3);
+  MAKE_GENERATOR(gen);
+  tnsr::I<DataVector, Dim, Frame::Inertial> inertial_coords{
+      mesh.number_of_grid_points()};
+  Jacobian<DataVector, Dim, Frame::Logical, Frame::Inertial> jacobian{
+      mesh.number_of_grid_points()};
+
+  fill_with_random_values(make_not_null(&inertial_coords), make_not_null(&gen),
+                          make_not_null(&dist));
+  fill_with_random_values(make_not_null(&jacobian), make_not_null(&gen),
+                          make_not_null(&dist));
+
+  InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial> result{};
+
+  dg::metric_identity_jacobian(make_not_null(&result), mesh, inertial_coords,
+                               jacobian);
+
+  // To test, take numerical divergence.
+  tnsr::i<DataVector, Dim, Frame::Inertial> divergence_terms{
+      mesh.number_of_grid_points(), 0.0};
+  DataVector buffer(mesh.number_of_grid_points());
+  std::array<Mesh<1>, Dim> meshes_1d{};
+  for (size_t i = 0; i < Dim; ++i) {
+    gsl::at(meshes_1d, i) = mesh.slice_through(i);
+  }
+  const Matrix identity{};
+
+  for (size_t i_hat = 0; i_hat < Dim; ++i_hat) {  // logical index
+    auto diff_matrices = make_array<Dim>(std::cref(identity));
+    gsl::at(diff_matrices, i_hat) =
+        Spectral::differentiation_matrix(gsl::at(meshes_1d, i_hat));
+    for (size_t i = 0; i < Dim; ++i) {  // inertial index
+      apply_matrices(make_not_null(&buffer), diff_matrices,
+                     result.get(i_hat, i), mesh.extents());
+      divergence_terms.get(i) += buffer;
+    }
+  }
+
+  Approx local_approx = Approx::custom().epsilon(1.0e-12).scale(1.);
+  const DataVector expected{mesh.number_of_grid_points(), 0.0};
+  for (size_t i = 0; i < Dim; ++i) {
+    CHECK_ITERABLE_CUSTOM_APPROX(divergence_terms.get(i), expected,
+                                 local_approx);
+  }
+
+  // Now check with a map that the terms are computed correctly. This is not
```

Comment:
"the terms" -> "the terms of the inverse Jacobian (times the Jacobian determinent)" or similar